src/lib/common/topology_sampler.cpp

```diff
@@ -149,12 +220,255 @@ static void topology_timer_cb(void *arg) {
+ * @return True if the request was handled, false otherwise.
+ */
+static bool sys_info_reply_cb(bool ack, uint32_t msg_id, size_t service_strlen,
+                              const char *service, size_t reply_len, uint8_t *reply_data) {
+  if (ack) {
+    SysInfoSvcReplyMsg::Data reply = {0, 0, 0, 0, NULL};
+    if (SysInfoSvcReplyMsg::decode(reply, reply_data, reply_len) != CborNoError) {
+      printf("Failed to decode sys info reply\n");
+      return false;
+    }
+    printf("Service: %.*s\n", service_strlen, service);
+    printf("Reply: %" PRIu32 "\n", msg_id);
+    printf(" * Node id: %" PRIx64 "\n", reply.node_id);
+    printf(" * Git SHA: 0x%08" PRIx32 "\n", reply.git_sha);
+    printf(" * Sys config CRC: 0x%08" PRIx32 "\n", reply.sys_config_crc);
+    printf(" * App name: %.*s\n", reply.app_name_strlen, reply.app_name);
+    if (xQueueSend(_sys_info_queue, &reply, 0) != pdTRUE) {
+      printf("Failed to send sys info\n");
+    }
+  } else {
+    printf("NACK\n");
+  }
+  return true;
+}
 
-void topology_sampler_init(BridgePowerController *power_controller, cfg::Configuration* hw_cfg, cfg::Configuration* sys_cfg) {
+/*!
+ * @brief Callback function for cbor map reply, sends to queue for processing.
+ * 
+ * @param[in] ack True if the message was acknowledged, false otherwise.
+ * @param[in] msg_id The message id.
+ * @param[in] service_strlen The length of the service string.
+ * @param[in] service The service string.
+ * @param[in] reply_len The length of the reply data.
+ * @param[in] reply_data The reply data.
+ * @return True if the request was handled, false otherwise.
+ */
+static bool cbor_config_map_reply_cb(bool ack, uint32_t msg_id, size_t service_strlen,
+                                     const char *service, size_t reply_len,
+                                     uint8_t *reply_data) {
+  if (ack) {
+    ConfigCborMapSrvReplyMsg::Data reply = {0, 0, 0, 0, NULL};
+    if (ConfigCborMapSrvReplyMsg::decode(reply, reply_data, reply_len) != CborNoError) {
+      printf("Failed to decode cbor map reply\n");
+      return false;
+    }
+    printf("Service: %.*s\n", service_strlen, service);
+    printf("Reply: %" PRIu32 "\n", msg_id);
+    printf(" * Node id: %" PRIx64 "\n", reply.node_id);
+    printf(" * Partition: %" PRId32 "\n", reply.partition_id);
+    printf(" * Cbor map len: %" PRIu32 "\n", reply.cbor_encoded_map_len);
+    printf(" * Success: %" PRId32 "\n", reply.success);
+    if (reply.success) {
+      for (uint32_t i = 0; i < reply.cbor_encoded_map_len; i++) {
+        if (!reply.cbor_data) {
+          printf("NULL map\n");
+          break;
+        }
+        printf("%02x ", reply.cbor_data[i]);
+        if (i % 16 == 15) { // print a newline every 16 bytes (for print pretty-ness)
+          printf("\n");
+        }
+      }
+    }
+    printf("\n");
+    if (!xQueueSend(_config_cbor_map_queue, &reply, 0)) {
+      printf("Failed to send cbor map\n");
+    }
+  } else {
+    printf("NACK\n");
```

Comment:
Since we have multiple requests that might nack, print more info here to aid debugging.

Reply:
Added the service + msg id at the top of the function.